### PR TITLE
Feature/sw 1862 whmcs initiate au cor from whmcs domain module

### PR DESCRIPTION
**Pull request by ALameLlama**

### CODE REVIEW

File: modules/registrars/synergywholesaledomains/synergywholesaledomains.php [hunk 1]
```diff
@@ -1467,6 +1467,80 @@ function synergywholesaledomains_manageChildHosts(array $params)
...
+
+    // If renewal period and no Cors exists
+    if (!empty($_REQUEST['renewalLength']) && empty($vars['cor'])) {
+        $renewalLength = $_REQUEST['renewalLength'];
+        // If valid period create an invoice and add meta
+        if (array_key_exists($renewalLength, $vars['pricing'])) {
+            $invoiceData = [
+                'userid' => $params['userid'],
+                'itemdescription1' => "Initiate CoR for {$params['domain']}",
+                'itemamount1' => $vars['pricing'][$renewalLength]['register'],
```
Comment (Sn0wCrack) [inline modules/registrars/synergywholesaledomains/synergywholesaledomains.php]: Are we able to use the renewal pricing for this rather than register?

**Review by Sn0wCrack — CHANGES_REQUESTED** [review]

**Review by Sn0wCrack — APPROVED** [review]

File: modules/registrars/synergywholesaledomains/synergywholesaledomains.php [hunk 2]
```diff
@@ -1507,6 +1507,103 @@ function synergywholesaledomains_manageChildHosts(array $params)
     ];
 }
 
+/**
+ * Controller for the "Initiate CoR" page.
+ *
+ * @param array $params
+ * @return array
+ */
+function synergywholesaledomains_initiateAuCorClient(array $params): array
```
Comment (Luke-McMahon) [inline modules/registrars/synergywholesaledomains/synergywholesaledomains.php]:
```suggestion
function synergywholesaledomains_initiateAuCorClient(array $params)
```

File: modules/registrars/synergywholesaledomains/synergywholesaledomains.php [hunk 3]
```diff
@@ -2018,20 +2115,22 @@ function synergywholesaledomains_push(array $params)
...
 /**
- * Register our custom pages pages we want to display in the Client Area.
+ * Register our custom pages we want to display in the Client Area.
  *
  * @param array $params
- * @return mixed
+ * @return array
  */
-function synergywholesaledomains_ClientAreaCustomButtonArray(array $params)
+function synergywholesaledomains_ClientAreaCustomButtonArray(array $params): array
```
Comment (Luke-McMahon) [inline modules/registrars/synergywholesaledomains/synergywholesaledomains.php]:
```suggestion
function synergywholesaledomains_ClientAreaCustomButtonArray(array $params)
```

File: modules/registrars/synergywholesaledomains/hooks.php [hunk 4]
```diff
@@ -7,10 +7,13 @@
  * @license https://github.com/synergywholesale/whmcs-domains-module/LICENSE
  */
 
-// http://docs.whmcs.com/Editing_Client_Area_Menus
-use WHMCS\View\Menu\Item as MenuItem;
+
+use Illuminate\Database\Capsule\Manager as Capsule;
```
Comment (Luke-McMahon) [inline modules/registrars/synergywholesaledomains/hooks.php]:
```suggestion
```

duplicate - probably a bad merge

**Pull request merged**